model/src/main/kotlin/PackageCurationData.kt

```diff
@@ -107,7 +108,15 @@ data class PackageCurationData(
      * duplicates.
      */
     @JsonInclude(JsonInclude.Include.NON_NULL)
-    val sourceCodeOrigins: List<SourceCodeOrigin>? = null
+    val sourceCodeOrigins: List<SourceCodeOrigin>? = null,
+
+    /**
+     * User defined labels associated with this package. The labels are not interpreted by the core of ORT itself, but
+     * can be used in parts of ORT such as plugins, in evaluator rules or in reporter templates.
+     */
+    @JsonInclude(JsonInclude.Include.NON_EMPTY)
+    @JsonPropertyOrder(alphabetic = true)
+    val labels: Map<String, String> = emptyMap()
```

Comment:
I believe we need some discussion whether this needs to be nullable. Note that the use-case is different from e.g. `declaredLicenseMapping` as that doesn't have a corresponding field in `Package`. IMO this is more like the `sourceCodeOrigins` case.

Some related questions:
- Should a curation be able to clear labels that come from another (previously applied) curation provider?
- Should non-conflicting keys accumulate / maps be merged? E.g. if one curation provider sets `foo=bar` and another `a=b` for the same package?

In my opinion maps should merge and maintain non-conflicting keys, and for conflicting keys the one applied last wins.

Reply:
> In my opinion maps should merge and maintain non-conflicting keys, and for conflicting keys the one applied last wins.

Ok, I see that you've added a test to confirm exactly that behavior 👍🏻 However, I believe that the `diff` implementation needs adjustment to follow the same semantics and produce a diff for the label entries.

Reply:
In the latest dev meeting discussion IIRC the agreement was that 
1. labels for same key do not accumulate, but get overwritten.

Regarding clearing, I believe this is not needed and would be much more difficult to implement.



> Ok, I see that you've added a test to confirm exactly that behavior 👍🏻 However, I believe that the `diff` implementation needs adjustment to follow the same semantics and produce a diff for the label entries.

Hm, I'm not sure this is possible to do. It's at minimum complex and I believe the results of the diff are currently not used.

Reply:
I changed the implementation to how I understood your proposal.